### PR TITLE
7008 - Add ability to change the header color in datagrid

### PR DESCRIPTION
**Pull request by ericangeles**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

This PR adds ability to set either `dark` or `light` color of the header in datagrid. By default, it is set to `dark` (for the meantime).

**Related github/jira issue (required)**:

Closes #7008 

**Steps necessary to review your pull request (required)**:
- Pull this branch, build, and run the app
- Go to http://localhost:4000/components/datagrid/example-header-background-color.html
- Check the example page setting, the `headerBackgroundColor` should be set to `light`
- The header of the datagrid should be light
- Check the figma file design and compare it to the example page https://www.figma.com/file/4CtBGz8dwvSa6JsBcdtVZQ/Data-Grid?node-id=1490%3A6082&t=eCTYpZWPtiIfxkz3-0
- Test the states in header e.g., focus and hover
- Play around with the test page and check any probable UI issue/s
- Test the dark in the index test page http://localhost:4000/components/datagrid/example-index
- Dark header should still the same as before

**Included in this Pull Request**:
~~- [ ] An e2e or functional test for the bug or feature.~~
- [x] A note to the change log.

### CODE REVIEW

**Review by tjamesallen15 — APPROVED** [review]
Well I didn't see anything wrong and it looks good. I based the review on the figma.

**Review by tmcconechy — CHANGES_REQUESTED** [review]
Looks good but a few things..

1. try new - contrast mode - notice the borders should be the same color
Screen Shot 2022-12-14 at 2 14 27 PM
1.  in classic mode hovering the dropdowns seem to show the border but is too light (may have been like this before)

**Review by yohannahbautista — DISMISSED** [review]

File: src/components/datagrid/datagrid.js [hunk 1]
```diff
@@ -1655,6 +1657,11 @@ Datagrid.prototype = {
     }
 
     this.activeEllipsisHeaderAll();
+
+    // Set the color background header (light and dark)
+    (self?.settings?.headerBackgroundColor === 'light') ? //eslint-disable-line
+      (self?.headerRowLeft?.find('th').addClass('light'), self?.headerRow?.find('th').addClass('light'), self?.headerRowRight?.find('th').addClass('light')) :
```
[the remark this answers is not in the document]
Comment (ericangeles) [reply src/components/datagrid/datagrid.js:1663]: @tmcconechy `headerRowLeft` is in different table so it needs to add class there.

[the remark this answers is not in the document]
Comment (ericangeles) [reply src/components/datagrid/datagrid.js:1663]: @tmcconechy I didn't change anything in the border color in classic, looking at the older version, it seems the same.

[the remark this answers is not in the document]
Comment (tmcconechy) [reply src/components/datagrid/datagrid.js:1663]: Sure just looked off when testing. I would either do a quick fix or we can make an issue for later? Classic isn't really too important

**Review by tmcconechy — APPROVED** [review]
This is good for me now. Classic isnt important. Just need any tests you may want to fix (or skip)

**Comment by tmcconechy** [comment]
Oh @ericangeles can you pr the type for the new setting in NG - they will need that 👍🏻

**Review by deep7102 — APPROVED** [review]

**Comment by ericangeles** [comment]
@tmcconechy @deep7102 am skipping the visual tests as my docker's acting weird.

**Comment by tmcconechy** [comment]
Looks like just some lint stuff.

**Review by tmcconechy — APPROVED** [review]

**Pull request merged**